### DISCUSSION TITLE
Are there specific models of Emporia Vue 2 I should look to purchase if I want to flash the firmware?

### DISCUSSION THREAD

**Discussion by davehouser1** [emporia-vue-local/esphome#170] [General]
Hello, I am interested in getting a Emporia Vue 2, and flashing the firmware following the instructions this repo has to offer.

Question1: Are there different hardware versions of Emporia Vue 2 I should be aware of? I have flashed firmware on hardware before and some manufacturers of products may sell different hardware versions of their products. I want to make sure I buy a product that this site supports. I read through the README.md and didn't see anything about hardware version support, either way just want to make sure that any Emporia Vue 2 supports the flashing process.

Question2: if there is a specific hardware version I should be shopping for which one is it? I was planning on buying what's on amazon right now?

Question3: Is there a way to rollback the firmware to use Emporia's version incase things go wrong?

**Comment by ThailandCat** [comment]
1. Mine says v5 on PCB. I doubt you will be able to know which version you will buy, without opening the case which is not possible to do until you get product.
2. Same as above. I don't think there is any difference in firmware flashing. the device CPU is just ESP32 WROVER. Sample chip can be bought for $3. 
3. Yes, this is in documentation, practically first step you will do after soldering Serial header is to backup Emporia config into bin file. Then you can rollback it. Just do not do any initial setup before backup.
